vscode/src/chat/chat-view/handlers/ToolHandler.ts

```diff
@@ -0,0 +1,170 @@
+import type Anthropic from '@anthropic-ai/sdk'
+import type { ContentBlock, MessageParam, Tool, ToolResultBlockParam } from '@anthropic-ai/sdk/resources'
+import { PromptString } from '@sourcegraph/cody-shared'
+import type { SubMessage } from '@sourcegraph/cody-shared/src/chat/transcript/messages'
+import * as vscode from 'vscode'
+import type { AgentHandler, AgentHandlerDelegate, AgentRequest } from './interfaces'
+
+interface CodyTool {
+    spec: Tool
+    invoke: (input: any) => Promise<string>
+}
+
+interface ToolCall {
+    id: string
+    name: string
+    input: any
+}
+
+const allTools: CodyTool[] = [
+    {
+        spec: {
+            name: 'get_file',
+            description: 'Get the file contents.',
+            input_schema: {
+                type: 'object',
+                properties: {
+                    path: {
+                        type: 'string',
+                        description: 'The path to the file.',
+                    },
+                },
+                required: ['path'],
+            },
+        },
+        invoke: async (input: { path: string }) => {
+            // check if input is of type string
+            if (typeof input.path !== 'string') {
+                throw new Error(`get_file argument must be a string, value was ${JSON.stringify(input)}`)
+            }
+            const { path: relativeFilePath } = input
+            try {
+                const workspaceFolder = vscode.workspace.workspaceFolders?.[0]
+                if (!workspaceFolder) {
+                    throw new Error('No workspace folder found')
+                }
+                const uri = vscode.Uri.joinPath(workspaceFolder.uri, relativeFilePath)
+
+                const content = await vscode.workspace.fs.readFile(uri)
+                return Buffer.from(content).toString('utf-8')
```

Comment:
```suggestion
                return (await getContextFromRelativePath(filePath))?.content
```

We have a similar function that we might be able to reuse here